### ISSUE TITLE
ng test fails on angular CLI project

### ISSUE THREAD

**Issue by aecz** [pnpm/pnpm#863]
### Code to reproduce the issue:
```
pnpm i -g @angular/cli@1.2.5
ng new test-pnpm --skip-install
cd test-pnpm
pnpm i
ng test
```

### Expected behavior:
Tests are run successfully and start watching files.

### Actual behavior:
```
ng test

Cannot find module 'source-map-support'
Error: Cannot find module 'source-map-support'
    at Function.Module._resolveFilename (module.js:469:15)
    at Function.resolve (internal/module.js:27:19)
    at init (/Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/@angular/cli/1.2.5/node_modules/@angular/cli/plugins/karma.js:61:46)
    at Array.invoke (/Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/di/0.0.1/node_modules/di/lib/injector.js:75:15)
    at Injector.get (/Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/di/0.0.1/node_modules/di/lib/injector.js:48:43)
    at /Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/karma/1.7.0/node_modules/karma/lib/server.js:143:20
    at Array.forEach (native)
    at Server._start (/Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/karma/1.7.0/node_modules/karma/lib/server.js:142:21)
    at Injector.invoke (/Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/di/0.0.1/node_modules/di/lib/injector.js:75:15)
    at Server.start (/Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/karma/1.7.0/node_modules/karma/lib/server.js:103:18)
    at Promise (/Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/@angular/cli/1.2.5/node_modules/@angular/cli/tasks/test.js:35:25)
    at Class.run (/Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/@angular/cli/1.2.5/node_modules/@angular/cli/tasks/test.js:15:16)
    at Class.run (/Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/@angular/cli/1.2.5/node_modules/@angular/cli/commands/test.js:104:25)
    at resolve (/Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/@angular/cli/1.2.5/node_modules/@angular/cli/ember-cli/lib/models/command.js:273:20)
    at Class.validateAndRun (/Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/@angular/cli/1.2.5/node_modules/@angular/cli/ember-cli/lib/models/command.js:251:12)
    at Promise.resolve.then.then (/Users/aecz/test-pnpm/node_modules/.registry.npmjs.org/@angular/cli/1.2.5/node_modules/@angular/cli/ember-cli/lib/cli/cli.js:157:24)
npm ERR! Test failed.  See above for more details.
```

### Additional information:

 - `node -v` prints: 6.11.0
 - Windows, OS X, or Linux?: OS X Sierra

**Comment by zkochan** [comment]
I'll check tonight but I have a feeling that this is caused by the same issue as #857. I contributed a fix to [resolve](https://github.com/substack/node-resolve) which was published yesterday. It is turned off by default for now (see https://github.com/substack/node-resolve/pull/131) but will be turned on in next major version. That should solve a lot of issues that pnpm has in the ecosystem

**Comment by zkochan** [comment]
this issue is happening because `@angular/cli` tries to resolve `source-map-support` even though `source-map-support` is not in its dependencies.

It works with npm because it flattens dependencies and source-map-support is hoisted from `karma-source-map-support`.

We'll have to make a small PR to `@angular/cli` in order to fix this.

**Comment by aecz** [comment]
PR done. https://github.com/angular/angular-cli/pull/7191
I fear it can happen with other packages though. I have seen other packages telling you to update to npm 3+ because of the flat tree. Any option to have pnpm detect if a package needs to be flattened?

**Comment by zkochan** [comment]
Thanks for making the PR!

There is no such option in pnpm currently but I've been working on a tool that should do that [node-modules-check](https://github.com/pnpm/node-modules-check). I'll have to check if it works and publish it

**Comment by aecz** [comment]
Solved but now i get another error:

    test-pnpm$ ng test
    ERROR [config]: Error in config file!
    { Error: Cannot find module 'enhanced-resolve/lib/ModulesInRootPlugin'

Should i close this issue and open a new one?

**Comment by zkochan** [comment]
As you wish. I created a separate label for grouping these issues.

**Comment by aecz** [comment]
PR created. https://github.com/angular/angular-cli/pull/7385

**Comment by whyboris** [comment]
I just tried out `pnpm` on an Angular project and it gave the error:
`Cannot find module 'enhanced-resolve/lib/ModulesInRootPlugin'`
I added `"enhanced-resolve": "3.4.1",` to my `package.json` and the error is gone.
Thank you @aecz for being on top of this -- looks like the Angular team will address the error I had and I'll no longer need to manually add _enhanced-resolve_ to my project 👍

**Comment by zkochan** [comment]
@aecz do you have twitter? I want to mention you in an article (to say thank you). Otherwise I'll just drop a link to your github page

**Comment by aecz** [comment]
@zkochan i do not use my Twitter account. Github profile is fine. Thanks for the mention.

**Comment by aecz** [comment]
`Cannot find module 'source-map-support'` error solved with Angular CLI 1.3.2.
